app/src/main/java/org/dicekeys/app/openpgp/SignaturePacket.kt

```diff
@@ -0,0 +1,129 @@
+package org.dicekeys.app.openpgp;
+
+import com.google.common.io.ByteStreams
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters
+import org.bouncycastle.math.ec.rfc8032.Ed25519
+import java.security.MessageDigest
+
+class SignaturePacket(privateKey: ByteArray, private val timestamp: UInt, userIdPacket: UserIdPacket): Packet() {
+
+    private val secretPacket = SecretPacket(privateKey, timestamp)
+
+    private val privateKeyEd255119 = Ed25519PrivateKeyParameters(privateKey, 0)
+
+    override val pTag: Int
+        get() = 0x88
+
+    val hashedSubPackets : ByteArray by lazy {
+        val subpackets = ByteStreams.newDataOutput()
+
+        subpackets.writeByte(0x16) // length
```

Comment:
Another spot where a spec pointer would be good, even if it may be inferred by the file that imports this class, the reader won't necessarily see that here.  (I didn't, at least)